### PR TITLE
cam6_3_133: Hybrid PGF option in the SE dycore for WACCM-x and other misc updates

### LINKED ISSUE #681: Correction to WACCMX geometric height diagnostic
### Issue Type

Other (please describe below)

### Issue Description

Geometric height diagnostic (Z3GM) calculation is missing surface height. The following change is a proposed fix:
```
                   ! Might need geometric height on midpoints for output
                   !------------------------------------------------------------
                   if (hist_fld_active('Z3GM')) then
-                     r8tmp = phys_state(lchnk)%zm(i, k)
+                     r8tmp = phys_state(lchnk)%zm(i, k) + phis(i)/gravit
                      tempm(i, k) = r8tmp * (1._r8 + (r8tmp * rearth_inv))
                   end if
```
Changes answers only for Z3GM history field.

### Will this change answers?

No

### Will you be implementing this yourself?

Yes

### LINKED ISSUE #896: WACCMX ne30pg3 is unstable
### What happened?

WACCM-x is unstable with the default ne30pg3 topo file introduced in tag cam6_3_101 when running longer than 9 time steps.### 

### What are the steps to reproduce the bug?

```
 create_test SMS_Ld1.ne30_ne30_mg17.FX2000.derecho_intel.cam-outfrq1d
```

### What CAM tag were you using?

cam6_3_129

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

Yes, but I will need some help

### Extra info

_No response_

### PR DESCRIPTION
**Pull request by fvitt**
Introduce a hybrid PGF option in the SE dycore for WACCM-x to make WACCM-x consistent with PGF used in CAM in the troposphere and traditional PGF formulation above to solve stability issues with the SE dycore (fixes #896)

Set reasonable default PE layouts for WACCM(x) and CAMChem on derecho. This solves some of the regression test failures listed in #892

Implements PHIHM history field for waccmx -- High Latitude Electric Potential

Fixes #681

### CODE REVIEW

**Review by PeterHjortLauritzen — APPROVED** [review]
Thanks Francis!

File: src/dynamics/se/dycore/prim_advance_mod.F90 [hunk 1]
```diff
@@ -1211,7 +1210,7 @@ subroutine compute_and_apply_rhs(np1,nm1,n0,dt2,elem,hvcoord,hybrid,&
              pgf_term(:,:,2)=pgf_term(:,:,2) + &
                   cpair*T0*(grad_logexner(:,:,2)-grad_exner(:,:,2)/exner(:,:))
            end if
-         elseif (pgf_formulation==2) then
+         elseif (pgf_formulation==2.or.pgf_formulation==3) then
            pgf_term(:,:,1)  = density_inv(:,:)*grad_p_full(:,:,1,k)
            pgf_term(:,:,2)  = density_inv(:,:)*grad_p_full(:,:,2,k)
          else
```
Comment (cacraigucar) [inline src/dynamics/se/dycore/prim_advance_mod.F90:1216]: The error message needs to be updated to include 3

File: src/physics/cam/vertical_diffusion.F90 [hunk 2]
```diff
@@ -287,7 +287,7 @@ subroutine vertical_diffusion_init(pbuf2d)
   integer        :: nbot_eddy   ! Bottom interface level to which eddy vertical diffusion is applied ( = pver )
   integer        :: k           ! Vertical loop index
 
-  real(r8), parameter :: ntop_eddy_pres = 1.e-5_r8 ! Pressure below which eddy diffusion is not done in WACCM-X. (Pa)
+  real(r8), parameter :: ntop_eddy_pres = 1.e-7_r8 ! Pressure below which eddy diffusion is not done in WACCM-X. (Pa)
```
Comment (cacraigucar) [inline src/physics/cam/vertical_diffusion.F90:290]: Should this eventually be a namelist, or is this a one-time change?

File: src/dynamics/se/dycore/prim_advance_mod.F90 (hunk 1 above)
Comment (fvitt) [reply src/dynamics/se/dycore/prim_advance_mod.F90:1216]: Updated error message

File: src/physics/cam/vertical_diffusion.F90 (hunk 2 above)
Comment (fvitt) [reply src/physics/cam/vertical_diffusion.F90:290]: This is a one-time change

**Review by cacraigucar — APPROVED** [review]

**Review by nusbaume — APPROVED** [review]

**Pull request merged**